### PR TITLE
Fixing NFC example for Mifare card

### PR DESCRIPTION
**Pull request by Ellerbach**
- Fixing NFX example with Mifare card for both PN1850 and PN532

### CODE REVIEW

File: src/devices/Pn5180/samples/Program.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ Pn5180 Ft4222()
 
     Ft4222Spi ftSpi = new Ft4222Spi(new SpiConnectionSettings(0, 1) { ClockFrequency = Pn5180.MaximumSpiClockFrequency, Mode = Pn5180.DefaultSpiMode, DataFlow = DataFlow.MsbFirst });
 
-    using GpioController gpioController = new(PinNumberingScheme.Board, new Ft4222Gpio());
+    GpioController gpioController = new(PinNumberingScheme.Board, new Ft4222Gpio());
```
Comment (krwq) [inline src/devices/Pn5180/samples/Program.cs:118]: good catch!

File: src/devices/Pn5180/samples/Program.cs [hunk 2]
```diff
@@ -230,22 +230,22 @@ void TypeA()
...
                 else
                 {
-                    Console.WriteLine($"Error reading bloc: {block}, Data: {BitConverter.ToString(mifareCard.Data)}");
+                    Console.WriteLine($"Error reading bloc: {block}");
                 }
 
-                if (block % 4 == 3)
-                {
+                if (block % 4 == 3 && mifareCard.Data is object)
+                    {
```
Comment (krwq) [inline src/devices/Pn5180/samples/Program.cs:248]: indent is off (was good previously)

File: src/devices/Pn532/samples/Program.cs [hunk 3]
```diff
@@ -121,22 +121,22 @@ void ReadMiFare(Pn532 pn532)
...
-                if (ret >= 0)
+                if (ret >= 0 && mifareCard.Data is object)
                 {
                     Console.WriteLine($"Bloc: {block}, Data: {BitConverter.ToString(mifareCard.Data)}");
                 }
                 else
                 {
                     Console.WriteLine(
-                        $"Error reading bloc: {block}, Data: {BitConverter.ToString(mifareCard.Data)}");
+                        $"Error reading bloc: {block}");
```
Comment (krwq) [inline src/devices/Pn532/samples/Program.cs:136]: consider putting this on the same line

**Review by krwq — APPROVED** [review]
couple of nits, LGTM

File: src/devices/Pn5180/samples/Program.cs (hunk 1 above)
Comment (Ellerbach) [reply src/devices/Pn5180/samples/Program.cs:118]: Well, it took me couple of seconds when I ran the sample to understand that there was a problem somewhere :-)

**Comment by krwq** [comment]
Thanks @Ellerbach!

**Pull request merged**